infra/build/functions/target_experiment.py

```diff
@@ -82,19 +82,25 @@ def run_experiment(project_name, target_name, args, output_path, errlog_path,
   local_stacktrace_path = os.path.join(build.out, 'stacktrace/')
   fuzzer_args = ' '.join(args + [f'-artifact_prefix={local_artifact_path}'])
 
-  # Upload err.log.
+  # Upload JCC's err.log.
   if errlog_path:
-    compile_step_index = next((i for (i, d) in enumerate(steps)
-                               if '&& compile' in ' '.join(d['args'])), -1)
+    compile_step_index = -1
+    for i, step in enumerate(steps):
+      step_args = step.get('args', [])
+      if not step_args:
+        continue
+      if '&& compile' in ' '.join(step_args):
+        compile_step_index = i
+        break
```

Comment:
Would it be simpler to merge these two `if` blocks? E.g.,
```python
if step_args and '&& compile' in ' '.join(step_args):
```

Reply:
Also `''.join([])` returns `''`, which should suffice too.

Reply:
yep can remove this redundant check